packages/builtin/array.pony

```diff
@@ -200,6 +206,64 @@ class Array[A] is Seq[A]
     reserve(len)
     _size = len
 
+  fun box read_u8[B: (A & Real[B] val & U8) = A](offset: USize): U8 ? =>
+    """
+    Reads a U8 from offset. This is only allowed for an array of U8s.
+    """
+    if offset < _size then
+      _ptr._offset(offset)._convert[U8]()._apply(0)
+    else
+      error
+    end
+
+  fun box read_u16[B: (A & Real[B] val & U8) = A](offset: USize): U16 ? =>
+    """
+    Reads a U16 from offset. This is only allowed for an array of U8s.
+    """
+    let bytes = _element_size()
```

Comment:
If this function is only allowed for `Array[U8]`, why do we need `_element_size()` here? Won't it always be `1`? Or am I missing something?

Reply:
I'm not sure to what degree it will get optimized away at runtime - I'm mainly concerned with making sure the code as no more complex than it needs to be, for maintainability reasons.

So my vote would be to remove it, along with the multiplication operations below where the `bytes` local is used.

Reply:
Complexity removed.